src/main/java/org/kitteh/irc/client/library/feature/InteractionDefaults.java

```diff
@@ -0,0 +1,163 @@
+            // we enforce a requirement on null.
+            throw new IllegalArgumentException("defaultString may not be an empty string, use null instead.");
+        }
+        this.defaultString = defaultString;
+    }
+
+    public InteractionDefaults() {
+        this.defaultString = null;
+    }
+
+    /**
+     * This method will throw a RuntimeException if you try to define a value that
+     * is already defined. This often indicates a problem and #replaceDefault should be used
+     * instead.
+     *
+     * @param key The interaction to set a value for
+     * @param defaultString The default value to assume when nothing is provided for this Interaction
+     * @return Returns this to permit chaining
+     */
+    public InteractionDefaults setDefault(Interaction key, @Nullable String defaultString) {
+        if (this.defaults.get(key) != null) {
+            throw new RuntimeException("The default value for " + key.toString() + " is already set with value: " + defaults.get(key));
+        }
+
+        this.defaults.put(key, defaultString);
+
+        return this;
+    }
+
+    /**
+     *
+     * @param key The default Interaction key to obtain the default of
+     * @return The string of the default value, or the predefined defaultString
+     */
+    public String getDefault(Interaction key) {
+        String value = this.defaults.get(key);
+
+        if (value == null) {
+            return this.defaultString;
+        }
+
+        return value;
+    }
+
+    /**
+     * @return This method returns a Map with your defaults, and will fill all Interaction keys that were not already
+     *         defined.
+     */
+    public Map<Interaction, String> getDefaults() {
+        for (Interaction key : Interaction.values()) {
+            this.defaults.computeIfAbsent(key, value -> this.defaultString);
+        }
+
+        return defaults;
+    }
+
+    /**
+     * This method will throw a RuntimeException if you try to define a value that
+     * is not already defined. This often indicates a problem and #setDefault should be used
+     * instead.
+     *
+     * @param key The interaction to set a value for
+     * @param defaultString The default value to assume when nothing is provided for this Interaction
+     * @return Returns this to permit chaining
+     */
+    public InteractionDefaults replaceDefault(Interaction key, @Nullable String defaultString) {
+        if (this.defaults.get(key) == null) {
+            throw new RuntimeException("The default value for " + key.toString() + " is not set. It cannot be replaced.");
+        }
+
+        this.defaults.replace(key, defaultString);
+
+        return this;
+    }
+
+    /**
+     * @param message The message to set for the PART event.
+     * @return Returns this to permit chaining
+     */
+    public InteractionDefaults setPartMessage(String message) {
```

Comment:
Why the enum at all if you're going to break it down into these pieces? Easier to stick to the enum.